### ISSUE TITLE
I want to train my own dataset from scratch, but have some doubts

### ISSUE THREAD

**Issue by LKssssZz** [IDEA-Research/DAB-DETR#56]
First of all thank you for your team's work！

My own dataset has 3k training samples and I wish to train from scratch using res50, this facilitates subsequent changes to the different backbone network. I have some questions that I hope will be answered:

1. If I don't use your publicly available pretrained weights, will the network use the pretrained weights of res50 obtained on the imagenet dataset by default?

2. At present, I have performed preliminary training. When using the default hyperparameters, I use 4 img/GPU 2 to train 100 epochs, and the result is still basically 0. Do I need more training cycles? How much?

thanks for your reply

**Comment by SlongLiu** [comment]
Thanks for your questions.

1. Yes, it will. The code will download R50 in torchvision by default for training.
2. I suggest checking the log for more details, such as if the loss converges during training. Moreover, I suggest using COCO-pre-trained weights for training for a better result.

**Comment by LKssssZz** [comment]
Thank you for your reply and suggestions, I will do further experiments in the future.

**Comment by Luhui-Zhao** [comment]
> First of all thank you for your team's work！
> 
> My own dataset has 3k training samples and I wish to train from scratch using res50, this facilitates subsequent changes to the different backbone network. I have some questions that I hope will be answered:
> 
> 1. If I don't use your publicly available pretrained weights, will the network use the pretrained weights of res50 obtained on the imagenet dataset by default?
> 2. At present, I have performed preliminary training. When using the default hyperparameters, I use 4 img/GPU 2 to train 100 epochs, and the result is still basically 0. Do I need more training cycles? How much?
> 
> thanks for your reply

同学你好，我遇到了和你一样的问题，请问是否可以加个QQ，详细的交流一下呢？谢谢您！我的QQ是755476579

**Comment by LKssssZz** [comment]
同学你好，抱歉我目前仍旧有一些工作需要处理，所以抱歉抽不出时间添加同学你的qq，但是我可以给你一些我当时的思路。


1. DAB-DETR是当时我尝试学习的detr-like模型，我当时使用的数据集training约5K数据、test约为3K，想从头开始training DAB-DETR确实遇到了问题。
2. 我推荐同学阅读DAB的前置文章Conditional DETR并学习该模型（如果阅读过请忽略，这篇文章尝试从obj query直接估计2D采样点）。并且我从头训练Conditional DETR取得了不错的性能，但是在DAB-DETR中取得较差的结果。我猜测的原因有一下：1）直接在decoder layer中通过MLP生成4d-bbox可能比较难以优化，对于数据较少的情况（而Cond DETR直接估计2D中心点是比较容易优化的）。2）我记得这篇文章优化了温度系数，但是我记得自己在实验时使用它的温度系统难以优化，所以改回了原始文章的默认1。3）数据量较少时，我当时尝试过各种detr-like的文章代码，结果普遍不如CNN方法的sota，例如cascade rcnn等等。但是过去这么久了不知道现在怎么样了。


以上


917481832
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "IDEA-Research/DAB-DETR"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年4月16日(星期二) 下午4:11
***@***.***&gt;;
***@***.***&gt;;"State ***@***.***&gt;;
主题:&nbsp;Re: [IDEA-Research/DAB-DETR] I want to train my own dataset from scratch, but have some doubts (Issue #56)





  
First of all thank you for your team's work！
 
My own dataset has 3k training samples and I wish to train from scratch using res50, this facilitates subsequent changes to the different backbone network. I have some questions that I hope will be answered:
  
If I don't use your publicly available pretrained weights, will the network use the pretrained weights of res50 obtained on the imagenet dataset by default?
 
At present, I have performed preliminary training. When using the default hyperparameters, I use 4 img/GPU 2 to train 100 epochs, and the result is still basically 0. Do I need more training cycles? How much?
  
thanks for your reply
  
同学你好，我遇到了和你一样的问题，请问是否可以加个QQ，详细的交流一下呢？谢谢您！我的QQ是755476579
